ZelBack/src/services/messageHelper.js

```diff
@@ -0,0 +1,172 @@
+const bitcoinMessage = require('bitcoinjs-message');
+const bitcoinjs = require('bitcoinjs-lib');
+const zeltrezjs = require('zeltrezjs');
+const { randomBytes } = require('crypto');
+
+const log = require('../lib/log');
+
+/**
+ * Creates a message object.
+ *
+ * @param {object} data
+ *
+ * @returns {object} message
+ */
+function createDataMessage(data) {
+  const successMessage = {
+    status: 'success',
+    data,
+  };
+  return successMessage;
+}
+
+/**
+ * Creates a message object indicating success.
+ *
+ * @param {string} message
+ * @param {string} [name]
+ * @param {string} [code]
+ *
+ * @returns {object} success message
+ */
+function createSuccessMessage(message, name, code) {
+  const successMessage = {
+    status: 'success',
+    data: {
+      code,
+      name,
+      message,
+    },
+  };
+  return successMessage;
+}
+
+/**
+ * Creates a message indicating a warning.
+ *
+ * @param {string} message
+ * @param {string} [name]
+ * @param {string} [code]
+ *
+ * @returns {object} warning message
+ */
+function createWarningMessage(message, name, code) {
+  const warningMessage = {
+    status: 'warning',
+    data: {
+      code,
+      name,
+      message,
+    },
+  };
+  return warningMessage;
+}
+
+/**
+ * Creates a message indicating an error.
+ *
+ * @param {string} message
+ * @param {string} [name]
+ * @param {string} [code]
+ *
+ * @returns {object} error message
+ */
+function createErrorMessage(message, name, code) {
+  const errMessage = {
+    status: 'error',
+    data: {
+      code,
+      name,
+      message: message || 'Unknown error',
+    },
+  };
+  return errMessage;
+}
+
+/**
+ * Returns unauthorized error message.
+ *
+ * @returns {object} unauthorized error message
+ */
+function errUnauthorizedMessage() {
+  const errMessage = {
+    status: 'error',
+    data: {
+      code: 401,
+      name: 'Unauthorized',
+      message: 'Unauthorized. Access denied.',
+    },
+  };
+  return errMessage;
+}
+
+/**
+ * Verifies if the message was properly signed.
+ *
+ * @param {object} message
+ * @param {string} address
+ * @param {string} signature
+ * @param {string} strMessageMagic
+ * @param {string} checkSegwitAlways
+ *
+ * @returns {bool} isValid
+ */
+function verifyMessage(message, address, signature, strMessageMagic, checkSegwitAlways) {
```

Comment:
The verifyMessage and signMessage are unrelated to the creation of error/data messages.
They are more related to signing, verifying, id services. I think they shall be located in different file

Reply:
@TheTrunk I can move them to verification helper if that works?

Reply:
Yes that works 👍